internal/metric/prometheus.go

```diff
@@ -61,70 +64,85 @@ func captureOnePrometheusMetric(log logr.Logger, address, query, errorQuery stri
 	promAPI := promv1.NewAPI(c)
 
 	// Make sure Prometheus is ready
-	targets, err := promAPI.Targets(context.TODO())
+	lastScrapeTime, err := checkReady(ctx, promAPI, completionTime)
 	if err != nil {
 		return 0, 0, err
 	}
 
-	queryTime := completionTime
-	for _, target := range targets.Active {
-		if target.Health != promv1.HealthGood {
-			continue
-		}
-
-		if target.LastScrape.Before(completionTime) {
-			// TODO Can we make a more informed delay?
-			return 0, 0, &CaptureError{RetryAfter: 5 * time.Second}
-		}
-
-		if target.LastScrape.After(queryTime) {
-			queryTime = target.LastScrape
-		}
-	}
-
-	// If we needed to adjust the query time, log it so we have a record of the actual time being used
-	if !queryTime.Equal(completionTime) {
-		log.WithValues("queryTime", queryTime).Info("Adjusted completion time for Prometheus query")
-	}
-
 	// Execute query
```

Comment:
should we loop here?
```
for _, query := range []string{ query, errorquery } {
  if query == "" { continue }
  <query scalar>
  <adjust>
}
```

Reply:
Yeah, that's a fair point, otherwise when we actually start querying for error rates the timing could be off. The completion time probably needs to be adjusted during the loop as well so the error range matches the adjusted trial range if we decided that the final scrape was required. I'm going to make an issue to track this for now.